### PR TITLE
support superUser requests for teams, orgs, and downgrades

### PR DESCRIPTION
**Pull request by mattkrick**
TEST
- [ ] can downgrade as SU
- [ ] can view nested teams & orgs for users that aren't the viewer. test query:
```
query {
  viewer {
    teams {
      teamMembers {
        user {
          id
          organizations {
            id
          }
          teams {
            id
          }
        }
      }
    }
  }
}
```

### CODE REVIEW

File: src/server/graphql/mutations/downgradeToPersonal.ts [hunk 1]
```diff
@@ -66,18 +68,18 @@ export default {
...
     teamIds.forEach((teamId) => {
       // I can't readily think of a clever way to use the data obj and filter in the resolver so I'll reduce here.
-      // This is probably a smelly piece of code telling me I should be sending this per-userId or per-org
+      // This is probably a smelly piece of code telling me I should be sending this per-viewerId or per-org
       const teamData = {orgId, teamIds: [teamId]}
       publish(TEAM, teamId, DowngradeToPersonalPayload, teamData, subOptions)
     })
     // the count of this users tier stats just changed, update:
-    await sendSegmentIdentify(userId)
+    await sendSegmentIdentify(viewerId)
```
Comment (jordanh) [inline src/server/graphql/mutations/downgradeToPersonal.ts]: I don't think this should be `viewerId`. In fact, I failed you on the first review: this should be an array of all affected userIds (all of the userIds on the org just downgraded). The resulting identify events will propagate the new tier counts to all of our systems (HubSpot, Mixpanel, etc.)

File: src/server/graphql/types/User.js [hunk 2]
```diff
@@ -218,21 +219,33 @@ const User = new GraphQLObjectType({
...
     organization,
-    organizations: require('../queries/organizations').default,
+    organizations: {
+      description: 'Get the list of all organizations a user belongs to',
+      type: new GraphQLNonNull(new GraphQLList(new GraphQLNonNull(Organization))),
+      async resolve ({id: userId}, args, {authToken, dataLoader}) {
+        const userOrgs = await dataLoader.get('orgsByUserId').load(userId)
+        userOrgs.sort((a, b) => (a.name > b.name ? 1 : -1))
+        const viewerId = getUserId(authToken)
+        if (viewerId === userId || isSuperUser(authToken)) {
```
Comment (jordanh) [inline src/server/graphql/types/User.js]: 👍

File: src/server/graphql/types/User.js [hunk 3]
```diff
@@ -218,21 +219,33 @@ const User = new GraphQLObjectType({
...
-        let teamIds
-        if (viewerId === userId) {
-          teamIds = authToken.tms
-        } else {
-          const user = await dataLoader.get('users').load(userId)
-          teamIds = user.tms.filter((teamId) => authToken.tms.includes(teamId))
-        }
+        const user = await dataLoader.get('users').load(userId)
+        const teamIds =
+          viewerId === userId || isSuperUser(authToken)
```
Comment (jordanh) [inline src/server/graphql/types/User.js]: 👍

**Comment by jordanh** [comment]
I agree witcha, let's ship this as a 🔥🔧 – I'm happy to be the chap that pushes it to prod after we get it merged

File: src/server/graphql/mutations/downgradeToPersonal.ts (hunk 1 above)
Comment (mattkrick) [reply src/server/graphql/mutations/downgradeToPersonal.ts]: good catch

**Pull request merged**